mergin/client.py

```diff
@@ -134,24 +152,25 @@ def user_agent_info(self):
         return f"{self.client_version} ({platform.system()}/{system_version})"
 
     def _check_token(f):
+        """Wrapper for creating/renewing authorization token."""
         def wrapper(self, *args):
-            if (not self._auth_session or self._auth_session['expire'] < datetime.now(timezone.utc)) and self._auth_params:
-                self.login(self._auth_params['login'], self._auth_params['password'])
+            if self._auth_params:
+                if self._auth_session:
+                    # Refresh auth token if it expired or will expire very soon
+                    delta = self._auth_session["expire"] - datetime.now(timezone.utc)
+                    if delta.total_seconds() < 5:
+                        self.login(self._auth_params["login"], self._auth_params["password"])
+                else:
+                    # Create a new authorization token
+                    self.login(self._auth_params['login'], self._auth_params['password'])
```

Comment:
worth adding a log before self.login() that we do not have a token yet, so going to log in